integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/utils/ServerManager.java

```diff
@@ -0,0 +1,206 @@
+        FileManipulator.deleteDirectory(new File(extractedCarbonDir));
+        String extractDir = "carbontmp" + System.currentTimeMillis();
+        String baseDir = System.getProperty("basedir", ".") + File.separator + "target";
+        log.info("Extracting carbon zip file.. ");
+        (new ArchiveExtractor()).extractFile(serverZipFilePath, baseDir + File.separator + extractDir);
+        this.carbonHome = (new File(baseDir)).
+                getAbsolutePath() + File.separator + extractDir + File.separator + extractedCarbonDir;
+
+        return carbonHome;
+    }
+
+
+    /**
+     * Start server instance
+     *
+     * @param carbonHomePath carbon home path
+     * @throws IOException
+     */
+    public synchronized void startServer(String carbonHomePath) throws IOException {
+
+        File carbonHomeDirectory = new File(carbonHomePath);
+        String scriptName = getStartupScriptFileName(carbonHomePath);
+        String[] cmdArray;
+
+        if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")) {
+            carbonHomeDirectory = new File(carbonHomePath + File.separator + "bin");
+            cmdArray = new String[]{"cmd.exe", "/c", scriptName + ".bat"};
+            process = Runtime.getRuntime().exec(cmdArray, (String[]) null, carbonHomeDirectory);
+        } else {
+            cmdArray = new String[]{"sh", "bin/" + scriptName + ".sh"};
+            process = Runtime.getRuntime().exec(cmdArray, (String[]) null, carbonHomeDirectory);
+        }
+
+        errorStreamHandler = new ServerLogReader("errorStream", process.getErrorStream());
+        inputStreamHandler = new ServerLogReader("inputStream", process.getInputStream());
+        inputStreamHandler.start();
+        errorStreamHandler.start();
+
+        long startupWaitTimeOut = 60000L;
+
+        long time = System.currentTimeMillis() + startupWaitTimeOut;
+
+        while (true) {
+            if (this.inputStreamHandler.getOutput().contains("WSO2 Message Broker is started.")) {
+                log.info("Server started successfully.");
+                break;
+            } else if (System.currentTimeMillis() >= time) {
+                log.error("Server did not started properly within " + startupWaitTimeOut + " milliseconds.");
+            }
+        }
+    }
+
+    /**
+     * Stop server instance
+     *
+     * @throws IOException
+     */
+    public synchronized void stopServer() throws IOException {
+
+        if (process != null) {
+            long endTimeMillis = System.currentTimeMillis() + 10000;
+
+            while (process.isAlive()) {
+                process.destroy();
+                if (System.currentTimeMillis() > endTimeMillis) {
+                    process.destroyForcibly();
+                    log.info("process destroyed forcibly");
+                    break;
+                }
+            }
+
+            String catPidCommand = "cat " + carbonHome + File.separator + "carbon.pid";
```

Comment:
lets document that its temp method and this assumption that it will be replaced (hopefully) by something else.